src/lib/containers/markdown/MarkdownEditor.tsx

```diff
@@ -17,7 +22,62 @@ export const MarkdownEditor: React.FunctionComponent<MarkdownEditorProps> = ({
   const [currentTab, setCurrentTab] = useState<MarkdownEditorTabs>(
     MarkdownEditorTabs.WRITE,
   )
-  const [text, setText] = useState<string>()
+  const [text, setText] = useState<string>('')
+  const [selectionStart, setSelectionStart] = useState<number>(0)
+  const textAreaRef = useRef<HTMLTextAreaElement>(null)
+
+  useEffect(() => {
+    const input = textAreaRef.current
+    if (input) {
+      input.setSelectionRange(selectionStart, selectionStart)
+    }
+  }, [textAreaRef, selectionStart])
+
+  const handleCommands = (command: CommandListType) => {
+    const textVal = textAreaRef.current
+    if (textVal) {
+      const start = textVal.selectionStart
+      const end = textVal.selectionEnd
+      const selected = text.substring(start, end)
+      const first = text.substring(0, start)
+      const later = text.substring(end, text.length)
+
+      const openSyntax = MARKDOWN_COMMANDS_DATA[command].openSyntax
+      const closeSyntax = MARKDOWN_COMMANDS_DATA[command].closeSyntax
+
+      switch (command) {
+        case 'code': {
+          const newText = `${first}
+          ${openSyntax}
+          ${selected}
+          ${closeSyntax}
+          ${later}`
+            .split('\n')
+            .map(s => s.trim())
```

Comment:
I don't think trimming whitespace is right, this would turn 

```
{
  "hello": "world"
}
```

into 

```
``
{
"hello": "world"
}
``
```

What case were you trying to solve here?